src/js/angular/clustermanagement/directives/cluster-configuration/multi-region.directive.js

```diff
@@ -15,7 +23,197 @@ function MultiRegion() {
         },
         link: ($scope) => {
-            // not implemented
+            // =========================
+            // Private variables
+            // =========================
+            const subscriptions = [];
+
+            // =========================
+            // Public variables
+            // =========================
+            $scope.isAdmin = false;
+            $scope.topology = undefined;
+            $scope.addingTag = false;
+            $scope.TopologyState = TopologyState;
+
+            // =========================
+            // Public functions
+            // =========================
+            $scope.add = () => {
+                $scope.addingTag = true;
+            };
+
+            $scope.cancel = () => {
+                $scope.addingTag = false;
+            };
+
+            $scope.createTag = (tag) => {
+                const payload = {
+                    tag
+                };
+                $scope.setLoader(true);
+                return ClusterRestService.addCusterTag(payload)
+                    .then(() => {
+                        toastr.success($translate.instant('cluster_management.cluster_configuration_multi_region.created_tag', {tag}));
+                    }).catch((response) => {
+                        const msg = getError(response);
+                        toastr.error(msg, $translate.instant('cluster_management.cluster_configuration_multi_region.error.creating'));
+                    }).finally(() => {
+                        $scope.addingTag = false;
+                        $scope.setLoader(false);
+                    });
+            };
+
+            $scope.deleteTag = (tag) => {
+                ModalService.openSimpleModal({
+                    title: $translate.instant('cluster_management.cluster_configuration_multi_region.confirm.delete', {tag}),
+                    message: $translate.instant('cluster_management.cluster_configuration_multi_region.confirm.warning'),
+                    warning: true,
+                    backdrop: 'static',
+                    stopPropagation: true
+                }).result
+                    .then(() => {
+                        $scope.setLoader(true);
+                        return ClusterRestService.deleteClusterTag(tag);
+                    })
+                    .then(() => toastr.success($translate.instant('cluster_management.cluster_configuration_multi_region.deleted_tag', {tag})))
+                    .catch((response) => {
+                        if (response === 'cancel') {
+                            return;
+                        }
+                        const msg = getError(response);
+                        toastr.error(msg, $translate.instant('cluster_management.cluster_configuration_multi_region.error.deleting'));
+                    })
+                    .finally(() => $scope.setLoader(false));
+            };
+
+            $scope.enableSecondaryMode = () => {
```

Comment:
Split this method a bit, because it's quite complex now

Reply:
done